src/views/core/view_cmd_executor.cpp

```diff
@@ -2,10 +2,248 @@
 #include "gui.hpp"
 #include "pointers.hpp"
 #include "services/hotkey/hotkey_service.hpp"
+#include "util/string_operations.hpp"
 #include "views/view.hpp"
 
 namespace big
 {
+	static std::vector<std::string> current_suggestion_list;
+	static std::string command_buffer;
+	static std::string auto_fill_suggestion;
+	static std::string selected_suggestion;
+
+	bool does_string_exist_in_list(const std::string& command, std::vector<std::string> list)
+	{
+		auto found = std::find_if(list.begin(), list.end(), [&](const std::string& cmd) {
+			return cmd == command;
+		});
+		return found != list.end();
+	}
+
+	std::vector<std::string> deque_to_vector(std::deque<std::string> deque)
+	{
+		std::vector<std::string> vector;
+		for (auto& element : deque)
+		{
+			vector.push_back(element);
+		}
+		return vector;
+	}
+
+	static void add_to_last_used_commands(const std::string& command)
+	{
+		if (does_string_exist_in_list(command, deque_to_vector(g.cmd.command_history)))
+		{
+			return;
+		}
+
+		if (g.cmd.command_history.size() >= 10)
+		{
+			g.cmd.command_history.pop_back();
+		}
+
+		g.cmd.command_history.push_front(command);
+	}
+
+	std::string auto_fill_command(std::string current_buffer)
+	{
+		if (command::get(rage::joaat(current_buffer)) != nullptr)
+			return current_buffer;
+
+		for (auto [_, cmd] : g_commands)
```

Comment:
...

Reply:
The key is used for a secondary sanity check in this case. Ill just change the naming to avoid confusion.